analyzers/tests/SonarAnalyzer.Test/TestCases/ArrayPassedAsParams.CSharp12.cs

```diff
@@ -3,18 +3,23 @@
 const int a = 1;
 int[] array = [2, 3];
 
-_ = new MyClass(1, [1, 2, 3]);     // Noncompliant
-_ = new MyClass(1, []);            // Noncompliant
+var class1 = new MyClass(1, [1, 2, 3]);             // Noncompliant
+_ = new MyClass(1, []);                             // Noncompliant
 
 // repro for https://github.com/SonarSource/sonar-dotnet/issues/8510
-_ = new MyClass(1, [a, .. array]);    // Noncompliant FP
+_ = new MyClass(1, [a, .. array]);                  // Compliant
 
-_ = new MyClass2([1], [1, 2, 3]);  // Noncompliant
+_ = new MyClass2([1], [1, 2, 3]);                   // Noncompliant
 _ = new MyClass2([1, 2, 3], 1);
-_ = new MyClass3([1, 2, 3], [4, 5, 6]); // Noncompliant FP
-_ = new MyClass3([[1, 2, 3], [4, 5, 6]]); // Noncompliant
-_ = new MyClass3([[1, 2, 3], [4, 5, 6]]); // Noncompliant
+
+_ = new MyClass3([1, 2, 3], [4, 5, 6]);             // Noncompliant FP
+
+_ = new MyClass4(class1, new(1, [1, .. array]));    // Compliant
+_ = new MyClass4([class1, new(1, [1, .. array])]);  // Noncompliant, outer collection raises, despite the nested spread operator
+//               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This case is why I do `expression.Elements` instead of `expression.DescendantNodes()`